### DISCUSSION TITLE
Clipboard Extension

### DISCUSSION THREAD

**Discussion by AdamMady** [TurboWarp/extensions#659] [Show and tell]
Hey there I'm a new member to this discussion page so I may as well introduce myself, My name is Adam and I've been a scratcher for 5 years and 8 months! Most of my projects are made in the turbowarp offline editor and aren't stored online. I haven't used my gihub account at all since it was made I never really knew many extensions existed other than the box2d physics extenstion until I revisited turbowarp to start a new project then I suddenly discovered there's a whole bunch of extensions now so I looked at some code to see how it was made (I already understood javascript so it was easy to pick up) then sped through a tutorial to understand it more and now I'm here!

 I recently wanted to make a tool in turbowarp where it involved text so I thought to myself "Instead of using lists so users could copy the text manually I could try to find a Clipboard Extension" and I looked around trying to find one and I couldn't find anything that was related to copying I did find Sensing Plus's clipboard tool after hours of searching since it didn't have text saying that it has a clipboard block (I don't blame it for not) and it was enough to make the tool but I really felt like making an extension that was strictly made for Utilizing the clipboard so I had a go on making one. and here's what I made so far.
![image](https://github.com/TurboWarp/extensions/assets/43925377/85342bd5-da4e-464a-9ac5-e7e49455adb0)

What I'm looking for right now is some feedback on it and some new ideas I'm currently attempting to add a "When Text Pasted" event

Here's the code (only works unsandboxed): https://dpaste.com/D3QGAJX26

**Comment by BlueDome77** [comment]
btw you can upload the extension file directly in this comment you just have to change the file extension name to .txt (Example: Clipboard.txt)
